lib/net.c

```diff
@@ -838,10 +838,43 @@ check_port_mtu(struct gatekeeper_if *iface, unsigned int port_idx,
 	return 0;
 }
 
+static int
+check_port_cksum(struct gatekeeper_if *iface, unsigned int port_idx,
+	const struct rte_eth_dev_info *dev_info,
+	struct rte_eth_conf *port_conf, bool gatekeeper_server)
+{
+	if (ipv4_if_configured(iface) &&
+			(port_conf->txmode.offloads &
+			 DEV_TX_OFFLOAD_IPV4_CKSUM) &&
+			!(dev_info->tx_offload_capa &
+			DEV_TX_OFFLOAD_IPV4_CKSUM)) {
+		G_LOG(NOTICE, "net: port %hu (%s) on the %s interface doesn't support offloading IPv4 checksumming\n",
+			iface->ports[port_idx], iface->pci_addrs[port_idx],
+			iface->name);
+		port_conf->txmode.offloads &= ~DEV_TX_OFFLOAD_IPV4_CKSUM;
+		return -1;
+	}
+
+	if (!gatekeeper_server &&
+			(port_conf->txmode.offloads &
```

Comment:
Do we need to check this condition? It seems that this flag will be always set before if it's Grantor.

Reply:
As on my other comment, we shouldn't care with the test that leads to adding `DEV_TX_OFFLOAD_IPV4_CKSUM` to `port_conf->txmode.offloads` in `check_port_cksum()`, that is the job of `check_port_offloads()`. So we can safely remove `!gatekeeper_server` off of the test. In this case, we can even drop the parameter `bool gatekeeper_server` of `check_port_cksum()`.